File: challenges/Topics/script/number_counter_topic.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import Int64

counter = 0
pub = None

def number_callback(msg):
    global counter
    counter += msg.data
    newMsg = Int64()
    newMsg.data = counter
    pub.publish(newMsg)

if __name__ == '__main__':
    rospy.init_node('number_counter')

    sub = rospy.Subscriber("/number_publisher", Int64, number_callback)
    pub = rospy.Publisher("/number_counter", Int64, queue_size = 10)

    rate = rospy.Rate(2)

    rospy.spin()

File: challenges/src/Topics/number_counter_topic.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>

int counter = 0;
ros::Publisher pub;

void	callback_number(const std_msgs::Int64 &msg)
{
	std_msgs::Int64 incremented_message;

	counter += msg.data;
	incremented_message.data = counter;
	pub.publish(incremented_message);
}

int	main(int argc, char **argv)
{
	ros::init(argc, argv, "Number_Counter_Topic");
	ros::NodeHandle nh;
	ros::Subscriber sub = nh.subscribe("/number_publisher",
		 1000, callback_number);
	pub = nh.advertise<std_msgs::Int64>("/number_counter", 10);

	ros::spin();
}
